src/object_functions.c

```diff
@@ -0,0 +1,2016 @@
+	f32 sp5C;
+
+	Object *obj4C_obj;
+
+	Object_7C_80034860 *obj7C;
+	Object_64_80034860 *obj4C_obj64;
+
+	s8 sp4F; // Boolean
+	s8 sp4E;
+	f32 sp48;
+	f32 sp44;
+	f32 sp40;
+	f32 sp3C;
+	s32 sp38; // Boolean
+
+	
+    sp4F = FALSE;
+    sp5C = speed;
+    if (osTvType == TV_TYPE_PAL) {
+        sp5C *= 1.2;
+    }
+    sp40 = obj->x_position + (obj->x_velocity * sp5C);
+    sp44 = obj->y_position + (obj->y_velocity * sp5C);
+    sp48 = obj->z_position + (obj->z_velocity * sp5C);
+    sp3C = 9.0f;
+
+    func_80031130(1, &obj->x_position, &sp40, -1);
+    sp38 = FALSE;
+    func_80031600(&obj->x_position, &sp40, &sp3C, &sp4E, 1, &sp38);
+    if (sp38) {
+        obj->x_velocity = (sp40 - obj->x_position) / sp5C;
+        obj->y_velocity = (sp44 - obj->y_position) / sp5C;
+        obj->z_velocity = (sp48 - obj->z_position) / sp5C;
+    }
+    func_80011570(obj, obj->x_velocity * sp5C, obj->y_velocity * sp5C, obj->z_velocity * sp5C);
+    if (sp38) {
+        func_8003FC44(obj->x_position, obj->y_position - 36.0f, obj->z_position, 44, 0, 0.2, 0);
+        sp4F = TRUE;
+    }
+	
+    if (obj->unk78 > 0) {
+        obj->unk78 -= speed;
+    } else {
+        sp4F = TRUE;
+    }
+	
+    if (obj->unk4C->unk13 < 80) {
+        obj4C_obj = (Object*)obj->unk4C->unk0;
+        if (obj4C_obj && (obj4C_obj->behaviorId == 1)) {
+            obj4C_obj64 = obj4C_obj->unk64;
+            sp4F = TRUE;
+            if (obj4C_obj64->unk0 != -1) {
+                obj4C_obj64->unk187 = 1;
+            }
+            obj7C = (Object_7C_80034860 *)obj->unk7C.word;
+            if (obj7C) {
+                obj7C->unkC = 180;
+            }
+            sp4F = TRUE;
+            func_8003FC44(obj->x_position, obj->y_position - 36.0f, obj->z_position, 44, 17, 0.5, 0);
+        }
+    }
+    if (sp4F) {
+        gParticlePtrList_addObject(obj);
+    }
+}
+
+void obj_init_torch_mist(Object *obj, LevelObjectEntry_Torch_Mist *entry) {
+    f32 phi_f0 = (s32)(entry->unk9 & 0xFF);
+    if (phi_f0 < 10.0f) {
+        phi_f0 = 10.0f;
+    }
+    phi_f0 /= 64;
+    obj->scale = obj->header->scale * phi_f0;
+    obj->unk78 = entry->unk8;
+}
+
+void obj_loop_torch_mist(Object *obj, s32 speed) {
+    obj->unk18 += obj->unk78 * speed;
+}
+
+void obj_init_effectbox(Object *obj, u8 *entry) {
+}
+
+#ifdef NON_EQUIVALENT
+
+// Has regalloc & stack issues.
+
+f32 func_800707C4(s16);
+f32 func_800707F8(s16);
+Object **get_object_struct_array(s32 *count);
+
+void obj_loop_effectbox(Object *obj, s32 speed) {
+    Object **objList;
+    Object_3C_80034B74 *obj3C;
+    Object_64_80034B74 *curObj64;
+    s32 i;
+    s32 numberOfObjects;
+    f32 xDiff;
+    f32 yDiff;
+    f32 zDiff;
+    f32 temp0;
+    f32 temp1;
+    f32 temp2;
+    f32 temp3;
+    f32 temp4;
+    f32 temp5;
+    f32 tempf0;
+    u8 new_var2;
+
+    obj3C = obj->unk3C_a.unk3C;
+    objList = get_object_struct_array(&numberOfObjects);
+    temp0 = func_800707F8((((0, obj3C->unkB)) << 8) * (-1));
```

Comment:
```suggestion
    temp0 = func_800707F8((obj3C->unkB << 8) * (-1));
```
It doesn't look like this effects the diff, can we just do this? It's also terrifying, frankly.